README.md

```diff
@@ -15,194 +15,335 @@ generously donated thousands of hours in code development and upkeep.
+[Setting Up a Windows Development Environment for the Selenium .NET Language Bindings](https://jimevansmusic.blogspot.com/2020/04/setting-up-windows-development.html);
+This article is out of date, but it includes more detailed descriptions and screenshots that some people might find useful.
 
-If you plan to compile the
-[IE driver](https://github.com/SeleniumHQ/selenium/wiki/InternetExplorerDriver),
-you also need:
+<details>
+<summary>Click to see Current Windows Setup Requirements</summary>
+
+#### Option 1: Automatic Installation from Scratch
+This script will ensure a complete ready to execute developer environment. 
+(nothing is installed or set that is already present unless otherwise prompted) 
+
+1. Open Powershell as an Administrator
+2. Execute: `Set-ExecutionPolicy Bypass -Scope Process -Force` to allow running the script in the process
+3. Navigate to the directory you want to clone Selenium in, or the parent directory of an already cloned Selenium repo 
+4. Download and execute this script in the powershell terminal: [scripts/dev-environment-setup.ps1]`
+
+#### Option 2: Manual Installation
+1. Allow running scripts in Selenium in general:
+    ```
+    Set-ExecutionPolicy -ExecutionPolicy RemoteSigned
+    ```
+2. Enable Developer Mode:
+    ```
+    reg add "HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\AppModelUnlock" /t REG_DWORD /f /v "AllowDevelopmentWithoutDevLicense" /d "1"
+    ```
+3. Install [MSYS2](https://www.msys2.org/), which is an alternative shell environment that provides Unix-like commands
+    * Add bin directory to `PATH` environment variable (e.g., `"C:\tools\msys64\usr\bin"`)
+    * Add `bash.exe` location as the `BAZEL_SH` environment variable (e.g., `"C:\tools\msys64\usr\bin\bash.exe"`)
+4. Install the latest version of [Visual Studio Community](https://visualstudio.microsoft.com/vs/community/)
+    * Use the visual studio installer to modify and add the "Desktop development with C++" Workload
+    * Add Visual C++ build tools installation directory location to `BAZEL_VC` environment variable (e.g. `"C:\Program Files\Microsoft Visual Studio\2022\Community\VC"`)
+    * Add Visual C++ Build tools version to `BAZEL_VC_FULL_VERSION` environment variable (this can be discovered from the directory name in `"$BAZEL_VC\Tools\MSVC\<BAZEL_VC_FULL_VERSION>"`)
+5. Add support for long file names (bazel has a lot of nested directories that can exceed default limits in Windows)
+    * Enable Long Paths support with these 2 registry commands:
+    ```shell
+    reg add "HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Command Processor" /t REG_DWORD /f /v "DisableUNCCheck" /d "1"
+    reg add "HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\FileSystem" /t REG_DWORD /f /v "LongPathsEnabled" /d "1"
+    ```
+    * Allow Bazel to create short name versions of long file paths: `fsutil 8dot3name set 0`
+    * Set bazel output to `C:/tmp` instead of nested inside project directory:
```

Comment:
Should we just commit this to the repo?

Reply:
Can we have a conditional toggle based on whether the user is on windows? How do we know they have access to their `C:\` drive even if on Windows?

Reply:
Right, we cannot use conditionals on startup options, so disregard.